src/main/java/com/imsavva/checkers/server/gui/GraphicInterface.java

```diff
@@ -1,22 +1,115 @@
 package com.imsavva.checkers.server.gui;
 
+import com.imsavva.checkers.server.beans.Board;
+import com.imsavva.checkers.server.beans.Cell;
+import com.imsavva.checkers.server.beans.Figure;
+import com.imsavva.checkers.server.beans.Player;
+import com.imsavva.checkers.server.model.*;
+import com.imsavva.checkers.server.model.exceptions.GameException;
+import lombok.extern.slf4j.Slf4j;
+
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.*;
+
+@Slf4j
+public class GraphicInterface extends JFrame implements ActionListener, BoardPanelListener {
+    private transient GameModel game;
+    private transient Board board;
 
-public class GraphicInterface extends JFrame {
     public GraphicInterface() {
+        initGame();
+        initMenuBar();
         initUI();
+        game.startGame();
+    }
+
+    private void initGame() {
+        board = new Board();
+        PathChecker pathChecker = new UgolkiPathChecker(board);
+        Player player1 = new Player("Dummy Player");
+        Player player2 = new Player("Test Player");
+        WinCriteria winCriteria = new UgolkiWinCriteria();
+        game = new UgolkiGame(board, player1, player2, pathChecker, winCriteria);
     }
 
-    public void initUI() {
-        add(new BoardPanel().initComponents());
+    private void initMenuBar() {
+        JMenuBar menuBar = new JMenuBar();
+
+        JMenu fileMenu = new JMenu("File");
+        fileMenu.setMnemonic(KeyEvent.VK_F);
+        menuBar.add(fileMenu);
+
+        JMenuItem newGameMenuItem = new JMenuItem("New Game", KeyEvent.VK_N);
+        newGameMenuItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_N, InputEvent.CTRL_DOWN_MASK));
+        newGameMenuItem.addActionListener(this);
+        fileMenu.add(newGameMenuItem);
+
+        JMenuItem exitMenuItem = new JMenuItem("Quit", KeyEvent.VK_Q);
+        exitMenuItem.setAccelerator(KeyStroke.getKeyStroke(KeyEvent.VK_Q, InputEvent.CTRL_DOWN_MASK));
+        exitMenuItem.addActionListener(this);
+        fileMenu.add(exitMenuItem);
+
+        this.setJMenuBar(menuBar);
+    }
+
+    private void initUI() {
+        BoardPanel boardPanel = new BoardPanel();
+        boardPanel.addMoveListener(this);
+        boardPanel.initComponents();
+
+        add(boardPanel);
 
         pack();
         setResizable(false);
 
         setTitle("Corners Game");
         setLocationRelativeTo(null);
-        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+    }
+
+    public BoardPanelPiece getPieceAt(int x, int y) {
+        Cell cell = board.getCellAt(x, y);
+        return BoardPanelPiece.builder()
+                .square(new Point(x, y))
+                .isEmpty(cell.isEmpty())
+                .isWhite(cell.isEmpty() || cell.getFigure().getColor() == Figure.Color.WHITE)
+                .build();
+    }
+
+    public void movingPiece(String from, String to) {
+        try {
+            game.move(from, to);
```

Comment:
This is where the board notifies that a piece has been moved, we check for end game state here too.